### PR TITLE
create materialized with count(distinct) should not support at current version #755

### PR DESCRIPTION
**Pull request by xuzifu666**
create materialized with count(distinct) should not support at current version #755

### CODE REVIEW

**Comment by xuzifu666** [comment]
@kangkaisen @imay

File: fe/fe-core/src/main/java/com/starrocks/analysis/CreateMaterializedViewStmt.java [hunk 1]
```diff
@@ -198,6 +198,11 @@ public void analyzeSelectClause(boolean ignoreCast) throws AnalysisException {
                 FunctionCallExpr functionCallExpr = (FunctionCallExpr) selectListItemExpr;
                 String functionName = functionCallExpr.getFnName().getFunction();
                 MVColumnPattern mvColumnPattern = FN_NAME_TO_PATTERN.get(functionName.toLowerCase());
+                // current version not support count(distinct) function in creating materialized view
+                if (functionName.toLowerCase().equals(FunctionSet.COUNT) && functionCallExpr.toSqlImpl().toLowerCase().contains("distinct")){
```
[the remark this answers is not in the document]
Comment (xuzifu666) [reply fe/fe-core/src/main/java/com/starrocks/analysis/CreateMaterializedViewStmt.java:202]: ok

File: fe/fe-core/src/test/java/com/starrocks/analysis/CreateMaterializedViewStmtTest.java [hunk 2]
```diff
@@ -1157,6 +1157,54 @@ public void testDeduplicateMV(@Injectable SlotRef slotRef1,
 
     }
 
+
+    @Test
+    public void testCountDistinct(@Injectable SlotRef slotRef1,
```
Comment (wyb) [inline fe/fe-core/src/test/java/com/starrocks/analysis/CreateMaterializedViewStmtTest.java:1162]: There is no count distinct related ut

File: fe/fe-core/src/main/java/com/starrocks/analysis/CreateMaterializedViewStmt.java [hunk 3]
```diff
@@ -198,6 +198,11 @@ public void analyzeSelectClause(boolean ignoreCast) throws AnalysisException {
                 FunctionCallExpr functionCallExpr = (FunctionCallExpr) selectListItemExpr;
                 String functionName = functionCallExpr.getFnName().getFunction();
                 MVColumnPattern mvColumnPattern = FN_NAME_TO_PATTERN.get(functionName.toLowerCase());
+                // current version not support count(distinct) function in creating materialized view
+                if (functionName.toLowerCase().equals(FunctionSet.COUNT) && functionCallExpr.isDistinct()){
```
Comment (wyb) [inline fe/fe-core/src/main/java/com/starrocks/analysis/CreateMaterializedViewStmt.java:202]: Better not put it in the middle of mvColumnPattern check

**Review by kangkaisen — DISMISSED** [review]

File: fe/fe-core/src/main/java/com/starrocks/analysis/CreateMaterializedViewStmt.java (hunk 3 above)
Comment (imay) [reply fe/fe-core/src/main/java/com/starrocks/analysis/CreateMaterializedViewStmt.java:201]: Use ignorecaseEquals

Comment (imay) [reply fe/fe-core/src/main/java/com/starrocks/analysis/CreateMaterializedViewStmt.java:201]: And, is `sum(distinct)` supported? If it is not supported, why not disable count(distinct) here.

Comment (HangyuanLiu) [reply fe/fe-core/src/main/java/com/starrocks/analysis/CreateMaterializedViewStmt.java:201]: I agree, maybe adding a MVColumnCountPattern is a better way ?

**Review by Astralidea — CHANGES_REQUESTED** [review]
if user execute create materialized count(distict before .
then upgrade FE can not startup because RollupJobV2.java will throw IOException.

**Review by Astralidea — DISMISSED** [review]

**Review by Astralidea — DISMISSED** [review]

**Review by wyb — DISMISSED** [review]

**Review by Astralidea — DISMISSED** [review]

**Review by Astralidea — APPROVED** [review]

**Review by wyb — APPROVED** [review]

**Review by imay — APPROVED** [review]

**Pull request merged**